e2e-tests/src/rewards.rs

```diff
@@ -232,14 +241,72 @@ pub fn check_points<'a>(
-    non_reserved_members_for_session: &[AccountId],
-) -> Vec<&'a AccountId> {
-    non_reserved_members
+
+pub fn get_era_for_session<C: AnyConnection>(connection: &C, session: SessionIndex) -> EraIndex {
+    let session_first_block = get_session_first_block(connection, session);
+    get_era(connection, Some(session_first_block))
+}
+
+pub fn setup_validators(
+    config: &Config,
+) -> anyhow::Result<(EraValidators<AccountId>, CommitteeSeats, SessionIndex)> {
+    let root_connection = config.create_root_connection();
+    let era = get_current_era(&root_connection);
+    // we need to wait for at least era 1 since some of the storage items are not available at era 0
+    if era == 0 {
+        wait_for_at_least_era(&root_connection, 1)?;
+    }
+
+    let members: Vec<_> = get_validators_keys(config)
         .iter()
-        .filter(|account_id| !non_reserved_members_for_session.contains(account_id))
-        .collect::<Vec<_>>()
+        .map(account_from_keypair)
+        .collect();
+    let members_size = members.len();
+    let reserved_count = std::cmp::min(members_size / 2, 2);
+    let reserved_members = &members[0..reserved_count];
+    let non_reserved_members = &members[reserved_count..];
+
+    let seats = COMMITTEE_SEATS;
+
+    let session = get_current_session(&root_connection);
+    let network_validators = get_era_validators(&root_connection, session);
+    let first_block_in_session = get_session_first_block(&root_connection, session);
+    let network_seats = get_committee_seats(&root_connection, Some(first_block_in_session));
+
+    let era_validators = EraValidators {
+        reserved: reserved_members.to_vec(),
+        non_reserved: non_reserved_members.to_vec(),
+    };
+
+    if era_validators == network_validators && seats == network_seats {
+        // nothing to do here
+        return Ok((era_validators, seats, session));
+    }
+
+    change_validators(
+        &root_connection,
+        Some(reserved_members.into()),
+        Some(non_reserved_members.into()),
+        Some(seats),
+        XtStatus::Finalized,
+    );
+
+    wait_for_next_era(&root_connection)?;
+    let session = get_current_session(&root_connection);
+
+    let network_validators = get_era_validators(&root_connection, session);
+    let reserved: HashSet<_> = era_validators.reserved.iter().cloned().collect();
+    let network_reserved: HashSet<_> = network_validators.reserved.into_iter().collect();
+    let non_reserved: HashSet<_> = era_validators.non_reserved.iter().cloned().collect();
+    let network_non_reserved: HashSet<_> = network_validators.non_reserved.into_iter().collect();
+    let first_block_in_session = get_session_first_block(&root_connection, session);
+    let network_seats = get_committee_seats(&root_connection, Some(first_block_in_session));
+
+    assert_eq!(reserved, network_reserved);
+    assert_eq!(non_reserved, network_non_reserved);
+    assert_eq!(seats, network_seats);
```

Comment:
instead of panic consider returning Err variant

Reply:
if you follow the advice, you may find `ensure` macro useful

Reply:
I would leave it as is -  it's a part of a test and I don't a way to handle such Err beyond just panicking.

Reply:
ok Im fine with this, @pmikolajczyk41 feel free to resolve or escalate it ;)